OpenSim/Sandbox/MatlabScripts/muscleCurves.m

```diff
@@ -0,0 +1,259 @@
+        
+        % get a valid name or 'exit' 
+        nameIndex = find( cellfun(@(s) ~isempty(strmatch(string, s)), musclelist) == 1);
+        
+        if strmatch(string, 'exit');
+            stoploop = 1;
+            break
+        elseif isempty(nameIndex);
+            display('Muscle name not found.');
+        else  
+            name = 1;
+            musclename = char( musclelist(nameIndex) );
+        end
+    end   
+    %%%     
+        
+    if ~isempty(stoploop)
+        display('thanks for ending this program. See you Laterz!')
+        close all
+        clc
+        return
+    end
+        
+    %% Muscle Coordinate finder
+    %   Find the coordinate's that each muscle crosses. This is done by
+    %   examining the moment arm contribution of the muscle across all
+    %   coordinates. A muscle will contribute to any coodinate when the moment
+    %   arm is non-zero.
+    muscle = musclecoordinates( model , s, musclename);
+
+    %% get the force length curves of 
+
+    % Get the force length curves of the muscles
+
+    [fl_active,fl_passive] = getForceLength(model, s, muscle);
+
+        %% plot the results
+
+    fig = figure(1);
+
+    clf(fig)
+    
+    hold
+
+    % Make scatter plots fot eh active and passive components
+    scatter(fl_active(:,1),fl_active(:,2))
+    scatter(fl_passive(:,1),fl_passive(:,2))
+    % Limit the X axis from 0.4 ? 1.6    
+    xlim([0.4 1.6]);
+
+    hold off
+
+    % clear the muscle name. 
+    name =[];
+end
+
+
+
+
+end
+
+
+
+
+
+
+
+function muscle = musclecoordinates(model,state,muscleName)
+% Muscle Coordinate finder
+%   Find the coordinate's that each muscle crosses. This is done by
+%   examining the moment arm contribution of the muscle across all
+%   coordinates. A muscle will contribute to any coodinate when the moment
+%   arm is non-zero.
+
+import org.opensim.modeling.*      % Import OpenSim Libraries
+
+% get the muscleCoordinates type buy getting the concrete Class Name
+force = model.getMuscles().get(muscleName);
+
+% get a reference to the concrete muscle class in the model
+muscleType = char(force.getConcreteClassName() );
+eval(['muscle =' muscleType '.safeDownCast(force);'])
+
+% get a fresh matrix to dump coordinate values into 
+momentArm_aCoord =[];
+nCoord = model.getCoordinateSet.getSize();
+
+% iterate through coordinates, finding non-zero moment arm's
+for k = 0 : nCoord -1
+    % get a reference to a coordinate
+    aCoord = model.getCoordinateSet.get(k);
+    % compute the moment arm of the muscle for that coordinate given
+    % the state. 
+    momentArm_aCoord(k+1) = muscle.computeMomentArm(state,aCoord);
+end
+
+% round the numbers. This is needed because at some coordinates there
+% are moments generated at the e-18 level. This is most likely
+% numerical error. So to deal with this I round to 4 decimal points. 
+x = round((momentArm_aCoord*1000))/1000;
+% Find the coordinate index's that are non-zero
+muscCoord = find(x ~= 0)-1;
```

Comment:
Could just use `find()` to locate entries of momentArm_aCoord that are within/beyond whatever your tolerance is (i.e., without rounding explicitly).